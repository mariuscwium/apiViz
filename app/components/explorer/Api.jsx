import React from 'react';
export default class Api extends React.Component {
    state = {
        loading: true
    };
    componentDidMount() {
        this.props.addPromise(
            fetch(this.props.endpoint)
                .then(function(response) {
                    if (response.ok) {
                        return response.json();
                    } else {
                        return Promise.reject('Unable to fetch data');
                    }
                })

                .then(body => {
                    this.setState({ data: body, loading: false });
                })
        );
    }

    render() {
        return this.props.children({
            ...this.state
        });
    }
}

Api.defaultProps = {
    addPromise: () => {}
};
