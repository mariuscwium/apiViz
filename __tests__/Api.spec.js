import React from 'react';
import Api from '../app/components/explorer/Api.jsx';
import renderer from 'react-test-renderer';

describe('Testing Api component', () => {
    beforeEach(() => {
        fetch.resetMocks();
    });

    test('Calls the add promise prop', () => {
        fetch.mockResponseOnce(JSON.stringify({ test }));
        const addPromise = jest.fn();
        renderer.create(
            <Api endpoint="test" addPromise={addPromise}>
                {state => {
                    return null;
                }}
            </Api>
        );
        expect(addPromise).toHaveBeenCalled();
    });

    test('Displays films when mounted', async () => {
        fetch.mockResponseOnce(JSON.stringify({ title: 'A New Hope' }));

        let promise;
        const addPromise = newPromise => {
            promise = newPromise;
        };

        const component = renderer.create(
            <Api endpoint="test" addPromise={addPromise}>
                {state => {
                    if (state.data) {
                        return <div>{state.data.title}</div>;
                    } else {
                        return null;
                    }
                }}
            </Api>
        );

        await promise;
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
