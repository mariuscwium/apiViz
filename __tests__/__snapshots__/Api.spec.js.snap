// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Api component Displays films when mounted 1`] = `
<div>
  A New Hope
</div>
`;
